crates/ingress-kafka/src/consumer_task.rs

```diff
@@ -75,8 +72,8 @@ impl fmt::Display for KafkaDeduplicationId {
     }
 }
 
-impl DeduplicationId for KafkaDeduplicationId {
-    fn requires_proxying(subscription: &Subscription) -> bool {
+impl KafkaDeduplicationId {
+    pub(crate) fn requires_proxying(subscription: &Subscription) -> bool {
         matches!(
             subscription.sink(),
             Sink::Service {
```

Comment:
Why do we only proxy for `Service` type invocations and not also for virtual objects and workflows?

Reply:
Wouldn't this mean that for virtual object or workflows, the Kafka ingress can write to any partition?